### PR TITLE
task spring-07 done

### PR DESCRIPTION
**Pull request by less08**

### CODE REVIEW

File: src/main/java/com/dev/cinema/annotation/EmailValidatorConstraint.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+
+public class EmailValidatorConstraint implements ConstraintValidator<EmailValidator,String> {
+    public static final String REGEX = "[a-zA-Z]+";
+
+    @Override
+    public void initialize(EmailValidator constraintAnnotation) {
+    }
+
```
Comment (Max-Pochepets) [inline src/main/java/com/dev/cinema/annotation/EmailValidatorConstraint.java]: [checklist](https://mate-academy.github.io/jv-program-fulltime/04_spring/checklist/hw_07.html)

File: src/main/java/com/dev/cinema/config/SecurityConfig.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+            .password(getEncoder().encode("pass")).roles("USER");
+    }
+
+    protected void configure(HttpSecurity http) throws Exception {
+        http
+            .authorizeRequests()
+            .anyRequest().authenticated()
+            .and()
+            .formLogin()
+            .permitAll()
```
Comment (Max-Pochepets) [inline src/main/java/com/dev/cinema/config/SecurityConfig.java]: keep permissions on the same level with the endpoints

File: src/main/java/com/dev/cinema/model/dto/MovieRequestDto.java [hunk 3]
```diff
@@ -1,7 +1,13 @@
 package com.dev.cinema.model.dto;
 
+import javax.validation.constraints.Min;
+import javax.validation.constraints.NotNull;
+
 public class MovieRequestDto {
+    @NotNull
+    @Min(4)
```
Comment (Max-Pochepets) [inline src/main/java/com/dev/cinema/model/dto/MovieRequestDto.java]: Not `@Min`. 
`@Size`

**Review by Max-Pochepets — APPROVED** [review]

File: src/main/java/com/dev/cinema/annotation/PasswordValidatorConstraint.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+        this.password = constraintAnnotation.field();
+        this.repeatPassword = constraintAnnotation.fieldMatch();
+    }
+
+    public boolean isValid(Object value,
+                           ConstraintValidatorContext context) {
+        Object fieldValue = new BeanWrapperImpl(value).getPropertyValue(password);
+        Object fieldMatchValue = new BeanWrapperImpl(value).getPropertyValue(repeatPassword);
+
+        if (fieldValue != null) {
```
Comment (boroda4436) [inline src/main/java/com/dev/cinema/annotation/PasswordValidatorConstraint.java:22]: do not complicate if-else statements

**Review by boroda4436 — APPROVED** [review]

**Pull request merged**